pkg/minishift/hostfolder/hostfolder_test.go

```diff
@@ -17,54 +17,10 @@ limitations under the License.
 package hostfolder
 
 import (
-	"testing"
-
-	"github.com/docker/machine/libmachine/drivers"
-	"github.com/minishift/minishift/pkg/minikube/tests"
 	"github.com/stretchr/testify/assert"
-
-	instanceState "github.com/minishift/minishift/pkg/minishift/config"
+	"testing"
 )
 
-func setupMock() (*tests.SSHServer, *tests.MockDriver) {
-	mockSsh, _ := tests.NewSSHServer()
-	mockSsh.CommandToOutput = make(map[string]string)
-	port, _ := mockSsh.Start()
-
-	mockDriver := &tests.MockDriver{
-		Port: port,
-		BaseDriver: drivers.BaseDriver{
-			IPAddress:  "127.0.0.1",
-			SSHKeyPath: "",
-		},
-	}
-	return mockSsh, mockDriver
-}
-
-func setupHostFolder() *instanceState.HostFolder {
-	return &instanceState.HostFolder{
-		Name: "Users",
-		Type: "cifs",
-		Options: map[string]string{
-			"mountpoint": "",
-			"uncpath":    "//127.0.0.1/Users",
-			"username":   "joe@pillow.us",
-			"password":   "am!g@4ever",
-			"domain":     "DESKTOP-RHAIMSWIN",
-		},
-	}
-}
-
-func TestHostfolderIsMounted(t *testing.T) {
```

Comment:
Does this mean we are not testing CIFS implementation?

Reply:
Cifs needs to have samba setup or use Windows. This for now an issue to setup...

Reply:
We really never tested CIFS as such. This was mocked test and due to the changes of the code I removed it. 

I agree that there are more tests missing. There are several areas where I would like to add more tests, but as discussed I don't have the bandwidth for that right now. 

Generally,  I would start with a set of integration tests for this. This would allow to test the whole range of hostfolder commands end to end. The SSHFS host folders have no prerequistes and work on all OS types, so we can write a single set of integration tests. I think doing this would give most value for money. Basically go through the cycle of:

- minishift hostfolder list -> nothing there
- minishift hostfolder add ... -> host folder defined
- minishift hostfolder list -> host folder defined, but not mounted
- minishift hostfolder mount -> host folder gets mounted, runs some commands that the file system is mounted and writable, etc
- minishift hostfolder list -> host folder defined and mounted
- minishift hostfolder umount
- minishift hostfolder delete

All this should be easily doable in a single cmd-hostfolder.feature.

If we are talking tests, I think that's what really missing here.
cc @agajdosi

Reply:
Make sense @hferentschik

Reply:
* minishift hostfolder mount -> host folder gets mounted, runs some commands that the file system is mounted and writable, etc
  * minishift hostfolder mount -> host folder errors out with 'Mounted already'.